### PR TITLE
Moved DeveloperSettings value checks to model from presenter

### PR DESCRIPTION
**Pull request by himattm**
Addressing Issue #54

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
I'm not sure if renaming existing methods like 'isLeakCanayEnabled()' adds any value to project, but moving checks to model as there are already similar ones to them is definitely a good point. 

Travis build found this issue:
```
com.artemzin.qualitymatters.ui.presenters.DeveloperSettingsPresenterTest > changeLeakCanaryState_shouldEnableLeakCanaryAndNotifyView FAILED
    Wanted but not invoked:
    developerSettingsModelImpl.changeLeakCanaryState(
        true
    );
    -> at com.artemzin.qualitymatters.ui.presenters.DeveloperSettingsPresenterTest.changeLeakCanaryState_shouldEnableLeakCanaryAndNotifyView(DeveloperSettingsPresenterTest.java:220)
```

File: app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModelImpl.java [hunk 1]
```diff
@@ -79,6 +79,10 @@ public boolean isStethoEnabled() {
         return developerSettings.isStethoEnabled();
     }
 
+    public boolean hasStethoStateChanged(boolean enabled) {
```
Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/developer_settings/DeveloperSettingsModelImpl.java:82]: These methods look pretty strange I would say…

File: app/src/debug/java/com/artemzin/qualitymatters/ui/presenters/DeveloperSettingsPresenter.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void syncDeveloperSettings() {
     }
 
     public void changeStethoState(boolean enabled) {
-        if (developerSettingsModel.isStethoEnabled() == enabled) {
+        if (!developerSettingsModel.hasStethoStateChanged(enabled)) {
```
Comment (artem-zinnatullin) [inline app/src/debug/java/com/artemzin/qualitymatters/ui/presenters/DeveloperSettingsPresenter.java:44]: it sort of removes logic, but logic is really is just boolean comparison that makes model code weird…

**Review by artem-zinnatullin — COMMENTED** [review]
idk about that, sorry, @vanniktech what do you think?

**Comment by artem-zinnatullin** [comment]
Agreed with @piotrek1543

**Review by vanniktech — COMMENTED** [review]
It seems super weird that the model checks this now. I don't think there's any benefit form this.

**Pull request closed without merging**